sktime/transformations/series/difference.py

```diff
@@ -121,16 +126,52 @@ class Differencer(BaseTransformer):
         "capability:inverse_transform": True,
     }
 
-    def __init__(self, lags=1, drop_na=True):
+    VALID_NA_HANDLING_STR = ["drop_na", "keep_na", "fill_zero"]
+
+    def __init__(self, lags=1, drop_na=None, na_handling="drop_na"):
         self.lags = lags
         self.drop_na = drop_na
+        self.na_handling = self._check_na_handling(na_handling)
+        # note: internally, we will use self._na_handling
+        #   because we must never change input param saves for sklearn compatibility
+        #   and because we need to "translate" the old "drop_na" arg
+        #   this could, in certain cases, overwrite the self. parameter
+        #   and cause sklearn compatibility issues due to the point mentioned
+
+        translate_arg = {True: "drop_na", False: "keep_na"}
+
+        if drop_na is not None:
```

Comment:
following sklrean we should not have any logic in `__init__` just argument assignment, all logic should happen in fit

Reply:
I will open an issue that highlights the problem

Reply:
@lmmentel, issue opened here:
https://github.com/alan-turing-institute/sktime/issues/2573

I interpret the fact that you did not make a change request as that this is non-blocking.